docs/enhancements/on-prem-data-collection.md

```diff
@@ -0,0 +1,143 @@
+
+## Motivation
+
+Data helps us see how the product is being used, helps us to analyze trends, and to determine ways we can improve the product for our users. 
+
+### Goals
+
+1. Create a process to collect and send [event](https://github.com/openshift/assisted-service/tree/master/docs/dev/events.md) data from on-premises deployments of AI.
+2. Analyze and process on-premises data (by use of visualizations, dashboards, etc.) at Red Hat similarly to SaaS data.
+3. Analyze on-premises data both separately and together with SaaS data.
+
+
+### Non-Goals
+
+1. Implementing new data types to collect is out of the scope of this epic. This epic defines how and when the data already collected will be processed and sent.
+2. Define how SaaS should redesign their data pipeline (taken care of in [MGMT-11244](https://issues.redhat.com/browse/MGMT-11244)) is out of the scope of this epic. This epic only refers to on-premises deployments of AI.
+3. Define the creation and/or management of an event streaming instance such as Kafka and the endpoint this data will be hosted at (both taken care of in [MGMT-11244](https://issues.redhat.com/browse/MGMT-11244)).
+4. Collecting must-gather data (and logs) for debugging failed cluster installation is outside the scope of this epic. This epic is only concerned with [event](https://github.com/openshift/assisted-service/blob/6b8401d3b0f0ceea5a837f33c17b9b1cd5ec52a3/internal/events/api/event.go) data.
+5. Data formatting and handling by the event streaming service is out of the scope of this epic. The event streaming service should be able to process the data from all versions of the on-premises environment.
+
+
+## Proposal
+
+A new flow for sending data is to be added:
+When a cluster installation attempt finishes (regardless of success or failure), all of the event data for this cluster will be queried from the DB, packaged, and sent to the data hosting endpoint.
+
+Changes that need to be made in the `Assisted-Service` to support this:
+- Needs to know what the data hosting service endpoint is and how to authenticate into it
+- Needs code changes when an install is attempted to query data
+- Needs to know how to format the data to send to the data hosting service
+- Needs to expose an option for users to opt out of this feature
+
+Additional changes needed:
+- Documentation regarding terms and conditions?
+- Documentation about how to opt out
```

Comment:
Isn't it just that if configuration for a destination is missing, it is opted out?

Reply:
Do you mean a procedure similar to the OCP health monitoring one where removing a JSON entry from the pull secret will opt the user out? https://access.redhat.com/documentation/en-us/openshift_container_platform/4.10/html-single/support/index#insights-operator-new-pull-secret_opting-out-remote-health-reporting

Reply:
@celebdor isn't opting out done in "real time" (i.e. the user can at any time opt out)?

Reply:
This can be a configuration in the operator I'd imagine (or just a flag at the deployment level).
Not sure why it would need to be more complicated than that.

Reply:
Discussed during the meeting with the conclusion that we'll expose an environment variable to opt-out that the user may toggle, but we'll also check if they opted-out of telemetry and respect that setting.